src/components/Item/Item.jsx

```diff
@@ -0,0 +1,19 @@
+import React, { useState } from 'react';
+import { useRouteMatch, Link } from 'react-router-dom';
+import style from './Item.css';
+
+export default function Item({ item, setSelectedItem }) {
+  const { name } = item;
+
+  return (
+    <div
+      className={style.itemThumb}
+      value={name}
+      onClick={(e) => {
+        setSelectedItem(e.target.innerHTML);
```

Comment:
This certainly works, but I'd in courage you to think about a different way to achieve this functionality more in line with the nested route architecture of the rest of your app